File: veritabani/kullanicilar.py
# 👥 Swix Dashboard - Kullanicilar Login Kontrol
# Basit login fonksiyonu

import sys
import os
from hashlib import sha256

sys.path.append(os.path.dirname(os.path.dirname(__file__)))

from veritabani.sql_baglantisi import vt_sorgu

def login_kontrol(mail, sifre):
    """Login kontrol - mail VEYA kullanici_adi ve şifreyi kontrol et"""
    # Email veya kullanıcı adı ile kontrol et
    sorgu_mail = """
    SELECT id, kullanici_adi, isim, soyisim, mail, sifre, role, active, firma, organization
    FROM Kullanicilar 
    WHERE (mail = %s OR kullanici_adi = %s) AND active = 1
    """
    
    sonuc_mail = vt_sorgu(sorgu_mail, (mail, mail))
    
    if sonuc_mail:
        kullanici = sonuc_mail[0]
        sifre_veritabani = kullanici[5]
        
        # Şifre kontrolü - plaintext ve hash'lenmiş şifreleri destekle
        sifre_plaintext_eslesme = sifre_veritabani == sifre
        sifre_hash = sha256(sifre.encode()).hexdigest()
        sifre_hash_eslesme = sifre_veritabani == sifre_hash
        
        if sifre_plaintext_eslesme or sifre_hash_eslesme:
            # Dönen değerler: (id, kullanici_adi, isim, soyisim, mail, role, firma, organization)
            return (kullanici[0], kullanici[1], kullanici[2], kullanici[3], kullanici[4], kullanici[6], kullanici[8], kullanici[9])
    
    return None

# Test
if __name__ == "__main__":
    print("Login Test:")
    kullanici = login_kontrol("muhammed.guc@bilgeguc.com", "admin")
    if kullanici:
        print(f"Başarılı: {kullanici[1]} ({kullanici[3]})")
    else:
        print("Başarısız")

File: veritabani/kullanicilar.php
<?php
// 👥 Swix Dashboard - Kullanicilar Login Kontrol
// Python kullanicilar.py dosyasının PHP karşılığı
// Basit login fonksiyonu

require_once __DIR__ . '/sql_baglantisi.php';

/**
 * Login kontrol - mail VEYA kullanici_adi ve şifreyi kontrol et
 * @param string $mail Email veya kullanıcı adı
 * @param string $sifre Şifre
 * @return array|null Kullanıcı bilgileri veya null
 */
function login_kontrol($mail, $sifre) {
    // Email veya kullanıcı adı ile kontrol et
    $sorgu_mail = "
        SELECT id, kullanici_adi, isim, soyisim, mail, sifre, role, active, firma, organization
        FROM Kullanicilar 
        WHERE (mail = ? OR kullanici_adi = ?) AND active = 1
    ";
    
    $sonuc_mail = vt_sorgu($sorgu_mail, [$mail, $mail]);
    
    if ($sonuc_mail && count($sonuc_mail) > 0) {
        $kullanici = $sonuc_mail[0];
        $sifre_veritabani = $kullanici[5];
        
        // Şifre kontrolü - plaintext ve hash'lenmiş şifreleri destekle
        $sifre_plaintext_eslesme = ($sifre_veritabani === $sifre);
        $sifre_hash = hash('sha256', $sifre);
        $sifre_hash_eslesme = ($sifre_veritabani === $sifre_hash);
        
        if ($sifre_plaintext_eslesme || $sifre_hash_eslesme) {
            // Dönen değerler: [id, kullanici_adi, isim, soyisim, mail, role, firma, organization]
            return [
                $kullanici[0],  // id
                $kullanici[1],  // kullanici_adi
                $kullanici[2],  // isim
                $kullanici[3],  // soyisim
                $kullanici[4],  // mail
                $kullanici[6],  // role
                $kullanici[8],  // firma
                $kullanici[9]   // organization
            ];
        }
    }
    
    return null;
}

// Test
if (basename(__FILE__) == basename($_SERVER['PHP_SELF'])) {
    echo "Login Test:\n";
    $kullanici = login_kontrol("muhammed.guc@bilgeguc.com", "admin");
    if ($kullanici) {
        echo "Başarılı: {$kullanici[1]} ({$kullanici[3]})\n";
    } else {
        echo "Başarısız\n";
    }
}
